instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -0,0 +1,325 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+
+import httpcore
+import httpx
+import wrapt
+
+from opentelemetry import context
+from opentelemetry.instrumentation.httpx.version import __version__
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import (
+    http_status_to_status_code,
+    unwrap,
+)
+from opentelemetry.propagate import inject
+from opentelemetry.trace import SpanKind, Tracer, TracerProvider, get_tracer
+from opentelemetry.trace.span import Span
+from opentelemetry.trace.status import Status
+
+ResponseInfo = typing.Tuple[
+    int, typing.List[typing.Tuple[bytes, bytes]], typing.Iterable[bytes], dict,
+]
+NameCallback = typing.Callable[[str, str], str]
+SpanCallback = typing.Callable[[Span, ResponseInfo], None]
+URL = typing.Tuple[bytes, bytes, typing.Optional[int], bytes]
+Headers = typing.List[typing.Tuple[bytes, bytes]]
+
+
+def _get_tracer(
+    *, tracer_provider: typing.Optional[TracerProvider] = None
+) -> Tracer:
+    return get_tracer(
+        __name__,
+        instrumenting_library_version=__version__,
+        tracer_provider=tracer_provider,
+    )
+
+
+def _get_span_name(
+    method: str,
+    url: str,
+    *,
+    name_callback: typing.Optional[NameCallback] = None
+) -> str:
+    span_name = ""
+    if name_callback is not None:
+        span_name = name_callback(method, url)
```

Comment:
We are following a pre/post reqeust/response hook pattern now. Would be nice if that was used instead of name callback.

Reply:
I replaced both of the previous callbacks with the request/response hooks.